Source/Events.Handling/Builder/EventHandlersBuilder.cs

```diff
@@ -34,24 +25,70 @@ public EventHandlersBuilder(ILoggerFactory loggerFactory)
+            where TEventHandler : class
+        {
+            _builders.Add(new ConventionEventHandlerBuilder(typeof(TEventHandler)));
+            return this;
+        }
+
+        /// <summary>
+        /// Registers a <see cref="Type" /> as an event handler class.
+        /// </summary>
+        /// <typeparam name="TEventHandler">The <see cref="Type" /> that is the event handler class.</typeparam>
+        /// <param name="eventHandlerInstance">The <typeparamref name="TEventHandler"/>.</param>
+        /// <returns>The <see cref="EventHandlersBuilder" /> for continuation.</returns>
+        public EventHandlersBuilder RegisterEventHandler<TEventHandler>(TEventHandler eventHandlerInstance)
+            where TEventHandler : class
+        {
+            _builders.Add(new ConventionEventHandlerBuilder(eventHandlerInstance));
+            return this;
+        }
+
+        /// <summary>
+        /// Registers a <see cref="Type" /> as an event handler class.
+        /// </summary>
+        /// <param name="eventHandlerInstance">The event handler instance.</param>
+        /// <returns>The <see cref="EventHandlersBuilder" /> for continuation.</returns>
+        public EventHandlersBuilder RegisterEventHandler(object eventHandlerInstance)
+        {
+            _builders.Add(new ConventionEventHandlerBuilder(eventHandlerInstance));
+            return this;
+        }
+
+        /// <summary>
+        /// Build and registers event handlers.
         /// </summary>
         /// <param name="eventProcessors">The <see cref="IEventProcessors" />.</param>
         /// <param name="eventTypes">The <see cref="IEventTypes" />.</param>
         /// <param name="processingConverter">The <see cref="IEventProcessingConverter" />.</param>
+        /// <param name="container">The <see cref="IContainer" />.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory" />.</param>
         /// <param name="cancellation">The <see cref="CancellationToken" />.</param>
-        public void BuildAndRegister(IEventProcessors eventProcessors, IEventTypes eventTypes, IEventProcessingConverter processingConverter, CancellationToken cancellation)
+        public void BuildAndRegister(
+            IEventProcessors eventProcessors,
+            IEventTypes eventTypes,
+            IEventProcessingConverter processingConverter,
+            IContainer container,
+            ILoggerFactory loggerFactory,
+            CancellationToken cancellation)
         {
+            var logger = loggerFactory.CreateLogger<EventHandlersBuilder>();
             foreach (var builder in _builders)
             {
-                builder.BuildAndRegister(eventProcessors, eventTypes, processingConverter, cancellation);
+                var buildResult = builder.BuildAndRegister(eventProcessors, eventTypes, processingConverter, container, loggerFactory, cancellation);
+                if (!buildResult.Succeeded) logger.LogWarning(buildResult.Warnings.ToString());
```

Comment:
I think it would be easier to just log warnings if something goes wrong during building inside the builders, and if they don't succeed, they will let the dev know that the handler will not be registered. Then we don't actually care about the result here anymore.

Reply:
This should be resolved by latest commits